tests/datasource/fluent/test_fabric.py

```diff
@@ -0,0 +1,143 @@
+    FabricPowerBIDatasource,
+    _PowerBIAsset,
+)
+
+if TYPE_CHECKING:
+    from great_expectations.data_context import AbstractDataContext
+
+param = pytest.param
+
+LOGGER = logging.getLogger(__name__)
+
+PYTHON_VERSION: Version = Version(f"{sys.version_info.major}.{sys.version_info.minor}")
+
+_DUMMY_POWER_BI_DATASET_ID: Final[str] = "0993ddd1-f01d-4c8f-a64e-9c4e5b8905e8"
+
+
+@pytest.fixture
+def empty_data_context(empty_data_context: AbstractDataContext) -> AbstractDataContext:
+    return empty_data_context
+
+
+@pytest.fixture
+def patch_power_bi_datasource(monkeypatch: pytest.MonkeyPatch) -> None:
+    """
+    Monkeypatch the PowerBI datasource to enable datasource creation even without `semantic-link` installed
+    """
+    monkeypatch.setattr(FabricPowerBIDatasource, "test_connection", lambda _: True)
+
+
+@pytest.fixture
+def power_bi_datasource(
+    empty_data_context: AbstractDataContext, patch_power_bi_datasource: None
+) -> FabricPowerBIDatasource:
+    datasource = empty_data_context.sources.add_fabric_powerbi(
+        "my_power_bi_datasource",
+        dataset=_DUMMY_POWER_BI_DATASET_ID,
+        workspace=None,
+    )
+    return datasource
+
+
+@pytest.fixture
+def capture_reader_fn_params(
+    monkeypatch: pytest.MonkeyPatch,
+) -> Generator[tuple[list[tuple], list[dict]], None, None]:
+    """
+    Capture the `reader_options` arguments being passed to the `PandasExecutionEngine`.
+
+    Note this fixture is heavily reliant on the implementation details of `PandasExecutionEngine`,
+    should this change this fixture will need to change.
+    """
+    captured_args: list[tuple] = []
+    captured_kwargs: list[dict[str, Any]] = []
+
+    def reader_fn_spy(*args, **kwargs) -> pd.DataFrame:
+        LOGGER.info(f"reader_fn_spy() called with...\n{args}\n{kwargs}")
+        captured_args.append(args)
+        captured_kwargs.append(kwargs)
+        return pd.DataFrame.from_dict({"foo": ["bar"]})
+
+    monkeypatch.setattr(
+        great_expectations.core.batch_spec.FabricBatchSpec,
+        "get_reader_function",
+        lambda _: reader_fn_spy,
+        raising=True,
+    )
+    LOGGER.info("Patcehd core.batch_spec.FabricBatchSpec.reader_method")
+
+    yield captured_args, captured_kwargs
+
+
+@pytest.mark.unit
+class TestFabricPowerBI:
+    @pytest.mark.parametrize(
+        ["asset_type", "asset_kwargs"],
+        [
+            param("powerbi_dax", {"dax_string": "my_dax_string"}, id="dax min_args"),
+            param(
+                "powerbi_measure",
+                {
+                    "measure": "my_measure",
+                },
+                id="measure min_args",
+            ),
+            param(
+                "powerbi_measure",
+                {
+                    "measure": "my_measure",
+                    "groupby_columns": [("foo", "bar"), ("fizz", "buzz")],
+                    # TODO: require custom serde for keys that are tuples
+                    # "filters": {
```

Comment:
filters is not supported for now, right?

Reply:
They are supported, they just won't be serializable.
So they would work with an in-memory data-context but they will fail to serialize with a `Cloud` of `File` Data context.

```python
import great_expectations as gx

in_memory_ctx = gx.get_context(mode="ephemeral")
file_ctx = gx.get_context(mode="file")
cloud_ctx = gx.get_context(mode="cloud")
```

Reply:
we changed the tuples to regular strings. would that solve the serialization/deserialization issue?

Reply:
Yes